### PR TITLE
migrate voltage level with the new design refactoring

### PR DESCRIPTION
**Pull request by Walid-Sahnoun**
Signed-off-by: walid-sahnoun <walid.sahnoun@rte-france.com>

### CODE REVIEW

**Review by laurgarn — APPROVED** [review]

File: src/main/java/org/gridsuite/modification/server/modifications/VoltageLevelCreation.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package org.gridsuite.modification.server.modifications;
```
Comment (dbraquart) [inline src/main/java/org/gridsuite/modification/server/modifications/VoltageLevelCreation.java:1]: copyright header is missing in this new file

File: src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
+package org.gridsuite.modification.server.modifications;
```
Comment (dbraquart) [inline src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java:1]: copyright header is missing in this new file

File: src/main/java/org/gridsuite/modification/server/modifications/VoltageLevelCreation.java (hunk 1 above)
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/modification/server/modifications/VoltageLevelCreation.java:7]: ok

File: src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java (hunk 2 above)
Comment (Walid-Sahnoun) [reply src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java:7]: done

File: src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+    @Override
+    protected MatcherModificationInfos createMatcher(ModificationInfos modificationInfos) {
+        return MatcherVoltageLevelCreationInfos.createMatcherVoltageLevelCreationInfos((VoltageLevelCreationInfos) modificationInfos);
+    }
+
+    @Override
+    protected void assertNetworkAfterCreation() {
+        assertNotNull(getNetwork().getVoltageLevel("vlId"));
+    }
```
Comment (dbraquart) [inline src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java:70]: you can go deeper in network assertions, adding:
```
assertNotNull(getNetwork().getBusbarSection("bbs.nw"));
assertNotNull(getNetwork().getBusbarSection("bbs.ne"));
assertNotNull(getNetwork().getBusbarSection("bbs.sw"));
assertTrue(getNetwork().getSubstation("s2").getVoltageLevelStream().anyMatch(vl -> vl.getId().equals("vlId")));
```

File: src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    @Override
+    protected void assertNetworkAfterDeletion() {
+        assertNull(getNetwork().getVoltageLevel("vlId"));
+    }
```
Comment (dbraquart) [inline src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java:75]: add:
```
assertNull(getNetwork().getBusbarSection("bbs.nw"));
assertNull(getNetwork().getBusbarSection("bbs.ne"));
assertNull(getNetwork().getBusbarSection("bbs.sw"));
assertFalse(getNetwork().getSubstation("s2").getVoltageLevelStream().anyMatch(vl -> vl.getId().equals("vlId")));
```

**Review by dbraquart — CHANGES_REQUESTED** [review]
review ok
tests ok
But we can improve network assertions in unit tests

File: src/main/java/org/gridsuite/modification/server/modifications/VoltageLevelCreation.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+    private final VoltageLevelCreationInfos modificationInfos;
+
+    public VoltageLevelCreation(VoltageLevelCreationInfos modificationInfos) {
+        this.modificationInfos = modificationInfos;
+    }
+
+    @Override
+    public void apply(Network network, Reporter subReporter) {
+        ModificationUtils.getInstance().createVoltageLevelAction(modificationInfos, subReporter, network);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/modifications/VoltageLevelCreation.java:26]: Rename by `createVoltageLevel`

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java [hunk 6]
```diff
@@ -304,9 +302,17 @@ public EquipmentCreationEntity createSubstationEntity(String id, String name, Co
     }
 
     public VoltageLevelCreationEntity createVoltageLevelEntity(String id, String name, double nominalVoltage, String substationId,
-        List<BusbarSectionCreationEmbeddable> busbarSections,
-        List<BusbarConnectionCreationEmbeddable> busbarConnections) {
-        return new VoltageLevelCreationEntity(id, name, nominalVoltage, substationId, busbarSections, busbarConnections);
+        List<BusbarSectionCreationInfos> busbarSections,
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:305]: Remove this method
Use DTO.toEntity

File: src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java (hunk 3 above)
Comment (Walid-Sahnoun) [reply src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java:74]: done

File: src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java (hunk 4 above)
Comment (Walid-Sahnoun) [reply src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java:84]: done

File: src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+        return MatcherVoltageLevelCreationInfos.createMatcherVoltageLevelCreationInfos((VoltageLevelCreationInfos) modificationInfos);
+    }
+
+    @Override
+    protected void assertNetworkAfterCreation() {
+        assertNotNull(getNetwork().getVoltageLevel("vlId"));
+        assertNotNull(getNetwork().getBusbarSection("bbs.nw"));
+        assertNotNull(getNetwork().getBusbarSection("bbs.ne"));
+        assertNotNull(getNetwork().getBusbarSection("bbs.sw"));
+        assertTrue(getNetwork().getSubstation("s2").getVoltageLevelStream().anyMatch(vl -> vl.getId().equals("vlId")));
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java:73]: assertEquals(1, getNetwork().getSubstation("s2").getVoltageLevelStream().filter(vl -> vl.getId().equals("vlId")).count());

File: src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    @Override
+    protected void assertNetworkAfterDeletion() {
+        assertNull(getNetwork().getVoltageLevel("vlId"));
+        assertNull(getNetwork().getBusbarSection("bbs.nw"));
+        assertNull(getNetwork().getBusbarSection("bbs.ne"));
+        assertNull(getNetwork().getBusbarSection("bbs.sw"));
+        assertFalse(getNetwork().getSubstation("s2").getVoltageLevelStream().anyMatch(vl -> vl.getId().equals("vlId")));
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java:82]: assertEquals(0, getNetwork().getSubstation("s2").getVoltageLevelStream().filter(vl -> vl.getId().equals("vlId")).count());

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+    }
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java:84]: Remove empty line

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java [hunk 10]
```diff
@@ -304,9 +302,17 @@ public EquipmentCreationEntity createSubstationEntity(String id, String name, Co
     }
 
     public VoltageLevelCreationEntity createVoltageLevelEntity(String id, String name, double nominalVoltage, String substationId,
-        List<BusbarSectionCreationEmbeddable> busbarSections,
```
Comment (dbraquart) [inline src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:307]: You should remove this function - we dont want functions with ALL parameters.
Callers must use DTO.toEntity() instead.

File: src/main/java/org/gridsuite/modification/server/modifications/VoltageLevelCreation.java (hunk 5 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/modification/server/modifications/VoltageLevelCreation.java:26]: and make it not public

Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/modification/server/modifications/VoltageLevelCreation.java]: i renamed and i removed the public modifier

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java (hunk 6 above)
Comment (Walid-Sahnoun) [reply src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java]: ok i changed

File: src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java (hunk 7 above)
Comment (Walid-Sahnoun) [reply src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java:73]: done

File: src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java (hunk 8 above)
Comment (Walid-Sahnoun) [reply src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java:83]: done

File: src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java (hunk 9 above)
Comment (Walid-Sahnoun) [reply src/test/java/org/gridsuite/modification/server/modifications/VoltageLevelCreationTest.java:85]: done

**Review by dbraquart — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**